state/src/interface/mod.rs

```diff
@@ -179,24 +96,36 @@ impl State {
         Ok(self_)
     }
 
+    /// Fills in the configuration for the replication node with handles to
+    /// state-managed resources, using the network manager to drive the raft
+    /// network
+    pub fn fill_replication_config(
+        &self,
+        replication_config: &mut ReplicationNodeConfig<GossipRaftNetwork>,
+        network_outbound: NetworkManagerQueue,
+        raft_inbound: RaftMessageReceiver,
+    ) {
+        let network =
+            GossipRaftNetwork::new(network_outbound, raft_inbound, self.translation_map.clone());
+
+        self.fill_replication_config_with_network(replication_config, network)
+    }
```

Comment:
This business of creating one object, using it to edit a config, then creating another is pretty messy in my opinion. This is a fine pattern if we internalize this to the `state` crate. So I'd suggest we publish a helper to start the raft worker and return a state instance. Essentially what is done in the `core` crate, but defined here to encapsulate the coupling.

This will make it worlds easier for tests to interact with state.

Reply:
after thinking about it somewhere, I think it would be cleaner to just have `State::new` return the handles to the Raft worker & cancel channel than go through all this back and forth. I can rename the method so it doesn't imply that it only returns a `State` object.